walkman.el

```diff
@@ -306,13 +304,25 @@ ARGS is the arg list from transient."
    ("c" "Copy as curl" walkman-copy-as-curl)
    ("i" "Import curl command" walkman-curl-to-org)])
 
+(defvar org-mode)
+
+(defvar walkman-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "C-c C-'") #'walkman-transient)
+    (define-key map (kbd "C-c <C-return>") #'walkman-at-point)
+    map))
+
 ;;;###autoload
-(defun walkman-mode ()
-  "Add the walkman bindings to org mode map."
-  (interactive)
-  (define-key org-mode-map (kbd "C-c C-'") #'walkman-transient)
-  (define-key org-mode-map (kbd "C-c <C-return>") #'walkman-at-point)
-  (message "Activated walkman"))
+(define-minor-mode walkman-mode
+  "walkman-mode."
+  :init nil
+  :lighter " Walkman"
+  :keymap walkman-mode-map
+  (if walkman-mode
+      (unless org-mode
+        (user-error "Walkman-mode makes sense in org-mode only")
+        (walkman-mode -1))
```

Comment:
would this "scope" the walkman minor mode to org mode only? why is it preferred to adding to the org mode map?

Reply:
I missed the story.
Before my change, the function was set to `org-mode-map`. So I thought it was a meaningful package only in org-mode.

As a naming problem, it's strange that it has a `*-mode` suffix but is not Major-mode / Minor-mode.
Perhaps it should be reset as a function-like name as `walkman-setup`, or it might be enough to just include a sample configuration in a comment.

Reply:
I debated for a while to call it `walkman-setup`. I do like that too. Maybe it does make more sense indeed

Reply:
OK, I fix branch and force-push.